src/App.jsx

```diff
@@ -14,45 +17,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_ALPHABETICALLY = 'alphabetically';
+const SORT_LENGTH = 'length';
+
+function getPreparedGood(goods, { sortField, isReversed }) {
+  const prepeardeGoods = [...goods];
+
+  if (sortField) {
+    prepeardeGoods.sort((item1, item2) => {
+      switch (sortField) {
+        case SORT_ALPHABETICALLY:
+          return item1.localeCompare(item2);
+        case SORT_LENGTH:
+          return item1.length - item2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    prepeardeGoods.reverse();
+  }
+
+  return prepeardeGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGood(goodsFromServer,
+    { sortField, isReversed });
+
+  const handleReverseClick = () => {
+    setIsReversed(prev => !prev);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const handleResetClick = () => {
+    setIsReversed(false);
+    setSortField('');
+  };
+
+  const handleSortClick = () => {
+    setSortField(SORT_ALPHABETICALLY);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(
+            'button is-info',
+            {
+              'is-light': sortField !== SORT_ALPHABETICALLY,
+            },
+          )}
+          onClick={handleSortClick}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button is-success',
+            {
+              'is-light': sortField !== SORT_LENGTH,
+            },
+          )}
+          onClick={() => setSortField(SORT_LENGTH)}
```

Comment:
For sake of consistency I belive it'll be better to move this one in the separate callback as well)